config/configuration.go

```diff
@@ -155,15 +155,15 @@ func GetCORS() cors.Config {
 	if !config.AllowAllOrigins {
 		allowOriginsRegexp := viper.GetString("cors.AllowOriginsRegexp")
 		if allowOriginsRegexp != "" {
-			originsRegexp, err := regexp.Compile(allowOriginsRegexp)
+			originsRegexp, err := regexp.Compile(fmt.Sprintf("^(%s)$", allowOriginsRegexp))
 			if err == nil {
 				config.AllowOriginFunc = func(origin string) bool {
 					return originsRegexp.Match([]byte(origin))
 				}
 			}
+		} else if len(viper.GetStringSlice("cors.AllowOrigins")) > 0 {
```

Comment:
This would look nicer as
```
} else if allowOrigins := viper.GetStringSlice("cors.AllowOrigins"); len(allowOrigins) > 0 {
    config.AllowOrigins = allowOrigins
```

Reply:
agree, fixed